pkg/v3/flows/logtrigger.go

```diff
@@ -75,53 +101,118 @@ func (flow *LogTriggerEligibility) ProcessOutcome(_ ocr2keepersv3.AutomationOutc
 	return nil
 }
 
-func newRecoveryFlow(rs ResultStore, rn Runner, logger *log.Logger, configFuncs ...tickers.RetryConfigFunc) (service.Recoverable, Retryer) {
-	// create observer
-	// no preprocessors required for retry flow at this point
-	// leave postprocessor empty to start with
-	recoveryObserver := ocr2keepersv3.NewObserver(nil, nil, rn, ObservationProcessLimit)
-
-	// create retry ticker
-	ticker := tickers.NewRecoveryTicker(RecoveryCheckInterval, recoveryObserver, logger, configFuncs...)
+func newRecoveryProposalFlow(rs ResultStore, ms MetadataStore, rp RecoverableProvider, recoveryInterval time.Duration, logger *log.Logger, configFuncs ...tickers.ScheduleTickerConfigFunc) (service.Recoverable, Retryer) {
+	// items come into the recovery path from multiple sources
+	// 1. [done] from the log provider as UpkeepPayload
+	// 2. [done] from retry ticker as CheckResult
+	// 3. [done] from primary flow as CheckResult if retry fails
+	// 4. [todo] from timeouts of the result store
+
+	// the recovery observer doesn't do any processing on the identifiers
+	// so this function is just a pass-through
+	f := func(_ context.Context, ids ...ocr2keepers.UpkeepPayload) ([]ocr2keepers.UpkeepPayload, error) {
+		return ids, nil
+	}
 
-	// postprocess
-	post := postprocessors.NewEligiblePostProcessor(rs)
+	// the recovery observer is just a pass-through to the metadata store
+	// add postprocessor for metatdata store
+	post := postprocessors.NewAddPayloadToMetadataStorePostprocessor(ms)
+
+	// TODO: add preprocessor that filters out in-progress ids
```

Comment:
The coordinator should act primarily as a filter in the recovery flow and not push values. It can potentially push values, but let's avoid that as much as possible.